### PR TITLE
refactor: snmp cleanup

### PR DESCRIPTION
**Pull request by murrant**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

Move common code to turn values to an array to a function.
Many snmp functions become an alias,  we can remove those.
snmp_cache_oid_num() has a slight change in result, it doesn't add an extra [''] into the array, just straight $array[$oid] = $value now.
remove unused functions

### CODE REVIEW

File: includes/snmp.inc.php [hunk 1]
```diff
@@ -370,140 +346,29 @@ function snmpwalk_cache_multi_oid($device, $oid, $array, $mib = null, $mibdir =
...
-            $ifIndex               = $array[$slotport]['ifIndex'];
-            $array[$ifIndex][$oid] = $value;
-        }
-    }
-
-    return $array;
-}//end snmp_cache_slotport_oid()
-
-
-function snmp_cache_oid($oid, $device, $array, $mib = 0)
```
Comment (laf) [inline includes/snmp.inc.php]: This is used in `./includes/polling/test.inc.php` which whilst we don't actually tell people to use is something we should either bin off or make use of.

**Comment by laf** [comment]
Running at home to test.

**Comment by laf** [comment]
All good to go, just that one comment about snmp_cache_oid being used in includes/polling/test.inc.php

**Comment by murrant** [comment]
Yeah, I figured this would break stuff.  Need to document things.

**Pull request merged**